### PR TITLE
Underlying changes to support multi-venue feature in Pro

### PR DESCRIPTION
**Pull request by borkweb**
Key changes:

* Tweak - Change styling for Venue Blocks to constrain to a card-like style. [ECP-1540]
* Tweak - Adjust REST endpoints to support multiple venues during event creation and updates. [ECP-1540]
* Tweak - If multiple venues exist on an event, display them in the event editor. [ECP-1540]
* Fix - Ensure the "Add Organizer" button is visible in the classic editor if organizers have been set. [TEC-4729]
* Fix - Simplify data handling of venues within the block editor. [ECP-1540]

🎥 
* [Classic editor, Pro deactivated](https://www.loom.com/share/f08ff2cdd1d748b3a343e62d32d05107)
* [Classic editor, Pro activated + Pro toggling](https://www.loom.com/share/8d8247b6974044a9be3c9ebb0406c71c)
* [Block editor, Pro deactivated](https://www.loom.com/share/4281fe5dc32b41759ff7818e096e90cd)
* [Block editor, Pro activated + Pro toggling](https://www.loom.com/share/a3eead60d04d48168ed5acc4c9bd4816)

🎫 
* [ECP-1540]
* [TEC-4729]

Related: https://github.com/the-events-calendar/events-pro/pull/2288

[ECP-1540]: https://theeventscalendar.atlassian.net/browse/ECP-1540?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ECP-1540]: https://theeventscalendar.atlassian.net/browse/ECP-1540?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ECP-1540]: https://theeventscalendar.atlassian.net/browse/ECP-1540?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[TEC-4729]: https://theeventscalendar.atlassian.net/browse/TEC-4729?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ECP-1540]: https://theeventscalendar.atlassian.net/browse/ECP-1540?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ECP-1540]: https://theeventscalendar.atlassian.net/browse/ECP-1540?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[TEC-4729]: https://theeventscalendar.atlassian.net/browse/TEC-4729?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/modules/data/blocks/venue/reducers/blocks.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+ */
+import { combineReducers } from 'redux';
+import { uniq } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import * as types from './../types';
+
+export const byId = ( state = {}, action ) => {
```
Comment (bordoni) [inline src/modules/data/blocks/venue/reducers/blocks.js:12]: Docblock

File: src/modules/data/blocks/venue/reducers/blocks.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+					...newState,
+					[ id ]: state[ id ],
+				};
+			}, {} );
+		default:
+			return state;
+	}
+};
+
+export const allIds = ( state = [], action ) => {
```
Comment (bordoni) [inline src/modules/data/blocks/venue/reducers/blocks.js:35]: Docblock

File: src/modules/data/blocks/venue/reducers/blocks.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+		case types.ADD_BLOCK_VENUE:
+			return uniq( [ ...state, action.payload.venue ] );
+		case types.REMOVE_BLOCK_VENUE:
+			return state.filter( venue => venue !== action.payload.venue );
+		default:
+			return state;
+	}
+};
+
+export const core = ( state = {}, action ) => {
```
Comment (bordoni) [inline src/modules/data/blocks/venue/reducers/blocks.js:46]: Docblock

File: src/modules/blocks/event-venue/template.js [hunk 4]
```diff
@@ -3,8 +3,7 @@
  */
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
-import { isEmpty } from 'lodash';
-import { addressToMapString } from '@moderntribe/events/editor/utils/geo-data';
+import { isEmpty, isInteger } from 'lodash';
```
Comment (bordoni) [inline src/modules/blocks/event-venue/template.js:6]:
```suggestion
import { isEmpty, isInteger, get } from 'lodash';
```

File: src/modules/blocks/event-venue/template.js [hunk 5]
```diff
@@ -290,16 +332,95 @@ class EventVenue extends Component {
...
+		if (
+			! state.events ||
+			! state.events.details ||
+			! state.events.details[ venueId ] ||
+			! state.events.details[ venueId ].details
+		) {
+			return {};
+		}
+
+		return state.events.details[ venueId ].details;
```
Comment (bordoni) [inline src/modules/blocks/event-venue/template.js:359]:
```suggestion
		return get( state, `events.details[${venueId}].details`, {} );
```

File: src/functions/template-tags/venue.php [hunk 6]
```diff
@@ -145,6 +145,30 @@ function tribe_get_venue_id( $postId = null ) {
...
+	$venue_ids = [];
+
+	if ( Tribe__Events__Main::instance()->isEvent( $event_id ) ) {
+		$venue_ids = tribe_get_event_meta( $event_id, '_EventVenueID', false );
+
+		// Protect against storing array items that render false, such as `0`.
+		$venue_ids = array_filter( (array) $venue_ids );
+	}
+
+	return apply_filters( 'tec_get_venue_ids', $venue_ids, $event_id );
```
Comment (bordoni) [inline src/functions/template-tags/venue.php:169]: Docblock.

Comment (bordoni) [reply src/functions/template-tags/venue.php:169]: And cast as Array.

**Review by bordoni — CHANGES_REQUESTED** [review]

File: src/modules/blocks/event-venue/template.js [hunk 7]
```diff
@@ -290,16 +332,86 @@ class EventVenue extends Component {
...
-		return ! isEmpty( details );
+		const details = this.getVenueDetails();
+
+		return ! isEmpty( details ) && isInteger( this.getVenueId() );
+	}
+
+	/**
+	 * Whether or not the venue block is an authoritative one.
+	 *
+	 * Authoritative mean it is a block that is showing the venue that was explicitly selected for it rather
```
Comment (Camwyn) [inline src/modules/blocks/event-venue/template.js:405]:
```suggestion
	 * Authoritative means it is a block that is showing the venue that was explicitly selected for it rather
```

File: src/views/blocks/parts/venue.php [hunk 8]
```diff
@@ -11,31 +11,33 @@
  *
  * @version 5.0.1
  *
+ * @var bool $show_map_link Whether to show the map link or not.
+ * @var ?int $venue_id The ID of the venue to display.
  */
 
 if ( ! tribe_get_venue_id() ) {
```
Comment (Camwyn) [inline src/views/blocks/parts/venue.php:18]: We should pass `$venue_id` to this?

File: src/resources/postcss/events-admin.pcss [hunk 9]
```diff
@@ -292,12 +292,25 @@
 }
 
 .linked-post-wrapper {
+	tbody + tbody tr:first-child td {
+		padding-top: calc(var(--tec-spacer-4) + 2px);
```
Comment (Camwyn) [inline src/resources/postcss/events-admin.pcss:296]: Typically, when faced with this I'd ask the designer to pick 20px or 24px 😉

**Review by Camwyn — COMMENTED** [review]
a couple small things

File: src/resources/postcss/events-admin.pcss (hunk 9 above)
Comment (borkweb) [reply src/resources/postcss/events-admin.pcss:296]: I'll pick `20px`!

**Review by bordoni — APPROVED** [review]

**Pull request merged**